src/vcpkg/sourceparagraph.cpp

```diff
@@ -365,12 +365,12 @@ namespace vcpkg
     }
 
     ParseExpected<SourceControlFile> SourceControlFile::parse_control_file(
-        const std::string& origin, std::vector<Parse::Paragraph>&& control_paragraphs)
+        StringView origin, std::vector<Parse::Paragraph>&& control_paragraphs)
     {
         if (control_paragraphs.size() == 0)
         {
             auto ret = std::make_unique<Parse::ParseControlErrorInfo>();
-            ret->name = origin;
+            ret->name = origin.to_string();
```

Comment:
Can we add a `::vcpkg::operator=(std::string&, StringView sv)`? (possibly in a future PR)

Reply:
The assignment operator cannot be a nommember.  http://eel.is/c++draft/over.binary#over.ass-1